### PR TITLE
Added Upstash Redis Memory support

### PR DESCRIPTION
**Pull request by ayushjain6013**
Feature: #372

### CODE REVIEW

**Comment by ayushjain6013** [comment]
@HenryHengZJ 
I have tried adding support for upstash redis. Could see it coming in the workflow running locally but could not actually test it as i am unable to create a sample workflow because i do not have the paid subscription of GPT or any other language models. Can you help me test it?

File: packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        this.baseClasses = [this.type, ...getBaseClasses(BufferMemory)]
+        this.inputs = [
+            {
+                label: 'Base URL',
+                name: 'baseURL',
+                type: 'string',
+                default: 'redis://localhost:6379'
+            },
+            {
+                label: 'Token',
```
Comment (HenryHengZJ) [inline packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts:36]: Can we put the token to credential file?
Here is an example PR - https://github.com/FlowiseAI/Flowise/pull/882/files

File: packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        this.icon = 'upstash.svg'
+        this.category = 'Memory'
+        this.description = 'Summarizes the conversation and stores the memory in upstash Redis server'
+        this.baseClasses = [this.type, ...getBaseClasses(BufferMemory)]
+        this.inputs = [
+            {
+                label: 'Base URL',
+                name: 'baseURL',
+                type: 'string',
+                default: 'redis://localhost:6379'
```
Comment (HenryHengZJ) [inline packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts:33]: this default url is wrong, from the [docs](https://js.langchain.com/docs/modules/memory/integrations/upstash_redis) it has to be upstash url: `https://<your-url>.upstash.io`

File: packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts (hunk 1 above)
Comment (ayushjain6013) [reply packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts]: Acknowledged

File: packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts (hunk 2 above)
Comment (ayushjain6013) [reply packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts]: Acknowledged

**Comment by HenryHengZJ** [comment]
Tested on Upstash: 
image

image

**Review by HenryHengZJ — APPROVED** [review]
thanks @ayushjain6013 !

**Pull request merged**